src/collaboration-platform/GitHubCollaborationPlatform.ts

```diff
@@ -5,37 +5,134 @@ import type {
   PullRequestInfo,
 } from "../shared/types";
 import type { CollaborationPlatform } from "./CollaborationPlatform";
+import { getOctokit } from "./OctokitAuth";
+import nodegit from "nodegit";
+
+async function repoPathToOwnerAndRepo(
+  repoPath: string,
+): Promise<{ owner: string; repo: string }> {
+  const repoResult = await nodegit.Repository.open(repoPath);
+  // TODO: Handle remotes that are not named "origin"
+  const remoteResult = await repoResult.getRemote("origin");
+  const remoteUrl = remoteResult.url();
+  // Sample URLs:
+  // "https://github.com/taneliang/stack-attack"
+  // "git@github.com:taneliang/stack-attack.git"
+  // "git@github.com:taneliang/hello.world"
+  const [rawRepo, owner] = remoteUrl
+    .split("/")
+    .flatMap((s) => s.split(":"))
+    .reverse();
+  const repo = rawRepo.endsWith(".git")
+    ? rawRepo.substr(0, rawRepo.length - 4)
+    : rawRepo;
+  return { owner, repo };
+}
```

Comment:
@manyaagarwal this function should be moved to ```ConstructStacker``` to compute the owner and repo from the repoPath. Then we'll pass these two in this ```GitHubCollaborationPlatform``` class